Cargo.toml

```diff
@@ -54,10 +57,12 @@ approx = "0.5.0"
 [features]
 default = ["bls12381", "ff_dfl"]
 bls12381 = []
+bn254 = []
+ristretto255 = []
 ff_dfl = []
 c = ["lang-c"]
 lp = ["good_lp", "lp-solvers"]
-r1cs = ["bellman"]
+r1cs = ["bellman", "spartan", "merlin", "curve25519-dalek"]
```

Comment:
Great! I see that you followed our plan here.

---

Beyond this PR, we probably should do some feature clean-up soon. For features that we intend to be externally set (either when using CirC as a library, or when building one of its binaries), I'd propose the following principle: **each feature should correspond to *one* backend or frontend**.

Thus, we'll want to make `r1cs` an internal feature (implied by other features, and not exposed in `driver.py`). It will be implied by either of `bellman` and `spartan` (the latter will imply `merlin` and `curve25519-dalek`).

We don't need to make that change in this PR, but I bring it up b/c I'd like to know your opinion of this idea, Jess. Also, @kwantam.

Reply:
I like this idea. What about in the case when you might want to compile R1CS without sending it immediately to a "built-in" proving system (like count)? I guess you would need to specify a modulus anyway (in most cases). So it would be okay to have `r1cs` as well as features implying different fields (`bls12381`, `ristretto255`, etc) implied by `bellman`/`spartan` features. Then you could specify, for example, `bellman` even if you weren't proving, to say essentially "use the bellman modulus to create an R1CS instance". I'm not sure if this is more confusing though.